From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Stained Glass OS <ke7oxh@gmail.com>
Date: Wed, 14 Oct 2026 20:55:42 -0700
Subject: [PATCH] crypt32: unsupported store control types give
 ERROR_NOT_SUPPORTED (3275)

The file and registry store providers answered an unknown CertControlStore control type with FALSE and no error (and a FIXME); they now set ERROR_NOT_SUPPORTED as documented.

diff --git a/dlls/crypt32/filestore.c b/dlls/crypt32/filestore.c
index 3753283..9155277 100644
--- a/dlls/crypt32/filestore.c
+++ b/dlls/crypt32/filestore.c
@@ -191,7 +191,12 @@ static BOOL WINAPI CRYPT_FileControl(HCERTSTORE hCertStore, DWORD dwFlags,
             ret = TRUE;
         break;
     default:
-        FIXME("%ld: stub\n", dwCtrlType);
+        /* a provider that does not support a control type: zero and ERROR_NOT_SUPPORTED */
+        TRACE("%ld: not supported\n", dwCtrlType);
+#ifdef SG_MUTANT_CRYPT32
+        break;
+#endif
+        SetLastError(ERROR_NOT_SUPPORTED);
         ret = FALSE;
     }
     return ret;
diff --git a/dlls/crypt32/regstore.c b/dlls/crypt32/regstore.c
index c537e27..2e01c7f 100644
--- a/dlls/crypt32/regstore.c
+++ b/dlls/crypt32/regstore.c
@@ -679,7 +679,12 @@ static BOOL WINAPI CRYPT_RegControl(HCERTSTORE hCertStore, DWORD dwFlags,
         break;
     }
     default:
-        FIXME("%lu: stub\n", dwCtrlType);
+        /* a provider that does not support a control type: zero and ERROR_NOT_SUPPORTED */
+        TRACE("%lu: not supported\n", dwCtrlType);
+#ifdef SG_MUTANT_CRYPT32
+        break;
+#endif
+        SetLastError(ERROR_NOT_SUPPORTED);
         ret = FALSE;
     }
     return ret;
